src/main/java/org/interledger/Condition.java

```diff
@@ -0,0 +1,66 @@
+package org.interledger;
+
+import java.util.Arrays;
+
+/**
+ * The execution condition attached to all transfers in an Interledger payment.
+ */
+public class Condition {
+
+  private byte[] hash;
```

Comment:
Let's make the properties `final` so that these objects are completely thread-safe.